core/txpool/celo_validation.go

```diff
@@ -0,0 +1,133 @@
+package txpool
+
+import (
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var NonWhitelistedFeeCurrencyError = errors.New("Fee currency given is not whitelisted at current block")
+
+// FeeCurrencyValidator validates currency whitelisted status at the specified
+// block number.
+type FeeCurrencyValidator interface {
+	IsWhitelisted(st *state.StateDB, feeCurrency *common.Address) bool
+	// Balance returns the feeCurrency balance of the address specified, in the given state.
+	// If feeCurrency is nil, the native currency balance has to be returned.
+	Balance(st *state.StateDB, address common.Address, feeCurrency *common.Address) *big.Int
+}
+
+func NewFeeCurrencyValidator() FeeCurrencyValidator {
+	return &feeval{}
+}
+
+type feeval struct {
+}
+
+func (f *feeval) IsWhitelisted(st *state.StateDB, feeCurrency *common.Address) bool {
+	// TODO: implement proper validation for all currencies
+	return feeCurrency == nil
+}
+
+func (f *feeval) Balance(st *state.StateDB, address common.Address, feeCurrency *common.Address) *big.Int {
+	// TODO: implement proper balance retrieval for fee currencies
+	return st.GetBalance(address)
+}
+
+// AcceptMap is a set of accepted transaction types for a transaction subpool.
+type AcceptMap = map[uint8]struct{}
+
+// CeloValidationOptions define certain differences between transaction validation
+// across the different pools without having to duplicate those checks.
+// In comparison to the standard ValidationOptions, the Accept field has been
+// changed to allow to test for CeloDynamicFeeTx types.
+type CeloValidationOptions struct {
+	Config *params.ChainConfig // Chain configuration to selectively validate based on current fork rules
+
+	AcceptMap AcceptMap // Map of transaction types that should be accepted for the calling pool
```

Comment:
Why do we have to go to a map? The previous bitmap approach seems ok (although less readable). If it is about allowing multiple tx types in one `Accept` bitmap, that should work with
```
Accept = 1 << txType1 + 1 << txType2
```

Reply:
Due to `CeloDynamicFeeTxType = 0x7b`. And I don't think we can change that type number.

Reply:
Now I see, and yes, we can't change the type and `1 << 0x7b` is slightly to big to fit into a uint8 ;-).

If we really wanted to keep the data structure, we could map out txtypes to the values 0-7. But your approach is clearer than introducing another mapping.